source/common/router/router.cc

```diff
@@ -737,14 +752,18 @@ void Filter::onResponseTimeout() {
     UpstreamRequestPtr upstream_request =
         upstream_requests_.back()->removeFromList(upstream_requests_);
 
-    // Don't record a timeout for upstream requests we've already seen headers
-    // for.
+    // Don't do work for upstream requests we've already seen headers for.
     if (upstream_request->awaiting_headers_) {
       cluster_->stats().upstream_rq_timeout_.inc();
       if (upstream_request->upstream_host_) {
         upstream_request->upstream_host_->stats().rq_timeout_.inc();
       }
 
+      // Record the max-value into the histogram to say that the entire timeout
+      // budget was used.
+      cluster_->timeoutBudgetStats().upstream_rq_timeout_budget_per_try_percent_used_.recordValue(
+          TimeoutPrecisionFactor);
+
```

Comment:
Is this always true? The global timeout might occur in the middle of a per-try timeout?